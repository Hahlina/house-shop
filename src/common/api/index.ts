export * from './fetchApi';
export * from './services/propety';
