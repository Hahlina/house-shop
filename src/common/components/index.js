export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
