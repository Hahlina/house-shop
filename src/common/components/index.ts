export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
