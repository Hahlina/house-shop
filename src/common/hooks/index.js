export * from './useProperties';
export * from './useProperty';
