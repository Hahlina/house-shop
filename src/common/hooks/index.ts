export * from './state';
export * from './useImageOnLoad';
