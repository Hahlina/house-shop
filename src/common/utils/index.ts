export * from './convertPrice';
export * from './capitalizeFirstLetter';
