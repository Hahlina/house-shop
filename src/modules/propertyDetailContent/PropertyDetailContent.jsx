import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { FullInformationCard, PropertyGallery, PropertyDetailContentLoader } from './components';

import { getProperty } from 'store/ducks';
import { EMPTY_OBJECT } from 'common/constants';
import { PropertyDetailContentStyled } from './PropertyDetailContent.styled';

export const PropertyDetailContent = () => {
    const { id } = useParams();
    const dispatch = useDispatch();
    const { data: properties, loading } = useSelector(({ propertyDetail }) => propertyDetail);
    const { photos, ...restInformation } = properties || EMPTY_OBJECT;

    useEffect(() => {
        dispatch(getProperty(id));
    }, [dispatch, id]);

    if (loading) return <PropertyDetailContentLoader />;

    return (
        <PropertyDetailContentStyled>
            <PropertyGallery {...{ photos }} />
            <FullInformationCard {...restInformation} />
        </PropertyDetailContentStyled>
    );
};
