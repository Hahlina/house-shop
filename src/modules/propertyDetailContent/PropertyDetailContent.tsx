import { useEffect } from 'react';
import { useParams } from 'react-router-dom';

import { Container, ErrorPage } from 'common/components';
import {
    FullInformationCard,
    PropertyGallery,
    PropertyDetailBreadCrumbs,
    PropertyDetailContentLoader
} from './components';

import { useAppDispatch, useAppSelector } from 'common/hooks';
import { getProperty, selectPropertyDetail } from 'store/ducks';
import { PropertyDetailContentStyled } from './PropertyDetailContent.styled';

export const PropertyDetailContent = () => {
    const { id } = useParams();
    const dispatch = useAppDispatch();
    const { data: property, loading, error } = useAppSelector(selectPropertyDetail);

    const { photos, ...restInformation } = property || {};

    useEffect(() => {
        id && dispatch(getProperty(id));
    }, [dispatch, id]);

    if (loading) return <PropertyDetailContentLoader />;
    if (error || !property) return <ErrorPage />;

    return (
        <Container>
            <PropertyDetailBreadCrumbs />
            <PropertyDetailContentStyled>
                {photos && <PropertyGallery {...{ photos }} />}
                {property && <FullInformationCard {...restInformation} />}
            </PropertyDetailContentStyled>
        </Container>
    );
};
