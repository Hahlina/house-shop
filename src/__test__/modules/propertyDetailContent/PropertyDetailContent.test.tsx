import { PropertyDetailContent } from 'modules/propertyDetailContent/PropertyDetailContent';

import { mockReduxStore, render, screen, mockPathName } from 'common/utils/test';
import * as reduxHooks from 'common/hooks';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Link: 'a',
    useParams: () => '1',
    useNavigate: jest.fn(),
    useLocation: () => () => ({ pathname: mockPathName })
}));

jest.mock('common/hooks');

describe('PropertyDetailContent page', () => {
    const mockedUseAppSelector = jest.spyOn(reduxHooks, 'useAppSelector');
    const mockedUseAppDispatch = jest.spyOn(reduxHooks, 'useAppDispatch');
    const { propertyDetail } = mockReduxStore;

    beforeEach(() => {
        mockedUseAppSelector.mockReturnValue(propertyDetail);
        mockedUseAppDispatch.mockReturnValue(jest.fn());
    });

    it('should render Container component', () => {
        render(<PropertyDetailContent />);
        const container = screen.getByTestId('container-id');
        expect(container).toBeInTheDocument();
    });

    it('should render title of FullInformationCard', () => {
        render(<PropertyDetailContent />);
        const propertyTitle = screen.getByText(propertyDetail.data.title);
        expect(propertyTitle).toBeInTheDocument();
    });

    it('should render PropertyGallery component', () => {
        render(<PropertyDetailContent />);
        const galleryContainers = screen.getAllByTestId('swiper-container');
        expect(galleryContainers).toHaveLength(2);
    });

    it('should render Breadcrumbs component', () => {
        render(<PropertyDetailContent />);
        const breadcrumbs = screen.getByLabelText('breadcrumbs');
        expect(breadcrumbs).toBeInTheDocument();
    });

    it('should take a snapshot', () => {
        const { container } = render(<PropertyDetailContent />);
        expect(container).toMatchSnapshot();
    });
});
