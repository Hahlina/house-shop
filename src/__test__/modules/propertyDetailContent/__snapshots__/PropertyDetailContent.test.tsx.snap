// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetailContent page should take a snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    data-testid="container-id"
  >
    <div
      class="sc-qZrbh bfdSXZ"
    >
      <nav
        aria-label="breadcrumbs"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              data-testid="breadcrumb-link"
              to="/"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 sc-hLQSwg fJWCBm css-ahj2mt-MuiTypography-root"
              >
                Home
              </p>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              to="#"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 sc-hLQSwg fJWCBm css-ahj2mt-MuiTypography-root"
              >
                Properties
              </p>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              to="/property/"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 sc-hLQSwg fJWCBm css-ahj2mt-MuiTypography-root"
              />
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="sc-fsYfdN huiuZN"
    >
      <div
        class="sc-eDLKkx bSMyoG"
      >
        <swiper-container
          class="sc-jTQCzO lnbGVf"
          data-testid="swiper-container"
          init="false"
        >
          <swiper-slide
            data-testid="swiper-slide"
          >
            <img
              alt="property-0"
              src="photo1.jpg"
            />
          </swiper-slide>
          <swiper-slide
            data-testid="swiper-slide"
          >
            <img
              alt="property-1"
              src="photo2.jpg"
            />
          </swiper-slide>
        </swiper-container>
        <swiper-container
          class="sc-gLLuof EQlPQ"
          data-testid="swiper-container"
          init="false"
        >
          <swiper-slide
            data-testid="swiper-slide"
          >
            <img
              alt="property-thumbs-0"
              src="photo1.jpg"
            />
          </swiper-slide>
          <swiper-slide
            data-testid="swiper-slide"
          >
            <img
              alt="property-thumbs-1"
              src="photo2.jpg"
            />
          </swiper-slide>
        </swiper-container>
      </div>
      <div
        class="sc-iBdnpw hdhuIM"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
        >
          Cozy House near the Forest
        </h5>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Address: 
          Green Street, 123
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Seller: 
          John Doe
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Description: 
          A charming house nestled near a beautiful forest. Perfect for nature lovers.
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 price css-ahj2mt-MuiTypography-root"
        >
          $250,000.00
        </p>
      </div>
    </div>
  </div>
</div>
`;
